Indicators/ExponentialMovingAverage.cs

```diff
@@ -15,48 +15,54 @@
 
 namespace QuantConnect.Indicators
 {
-
     /// <summary>
-    ///     Represents the traditional exponential moving average indicator (EMA)
+    /// Represents the traditional exponential moving average indicator (EMA)
     /// </summary>
-    public class ExponentialMovingAverage : Indicator
+    public class ExponentialMovingAverage : Indicator, IIndicatorWarmUpPeriodProvider
     {
         private readonly decimal _k;
-        private readonly int _period;
 
-        /// <summary>Initializes a new instance of the ExponentialMovingAverage class with the specified name and period
+        /// <summary>
+        /// Required period, in data points, for the indicator to be ready and fully initialized.
+        /// </summary>
+        public int WarmUpPeriod { get; }
+
+        /// <summary>
+        /// Initializes a new instance of the ExponentialMovingAverage class with the specified name and period
         /// </summary>
         /// <param name="name">The name of this indicator</param>
         /// <param name="period">The period of the EMA</param>
         public ExponentialMovingAverage(string name, int period)
             : base(name)
         {
-            _period = period;
-            _k = ExponentialMovingAverage.SmoothingFactorDefault(period);
+            WarmUpPeriod = period;
```

Comment:
Ema takes about 5x period to warm 
[ema-warm up.xlsx](https://github.com/QuantConnect/Lean/files/3074862/ema-warm.up.xlsx)
up

Reply:
I guided myself by the `IsReady` definition:
```csharp
public override bool IsReady => Samples >= _period;
```